connection.go

```diff
@@ -807,8 +807,8 @@ func (conn *Connection) reader(r io.Reader, c Conn) {
 			conn.reconnect(err, c)
 			return
 		}
-		resp := &Response{buf: smallBuf{b: respBytes}}
-		err = resp.decodeHeader(conn.dec)
+		buf := smallBuf{b: respBytes}
+		header, err := decodeHeader(conn.dec, &buf)
```

Comment:
> ```go
> conn.Do(req).GetReponse().Header().Error == iproto_ER_SOME_ERROR
> ```
> 
> This line will not work, because `GetReponse` returns `(Response, error)`.

Yeah, just an example that `Header().Error` works fine without `Decode()` call.

Reply:
Updated.